### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by karanJ2212**
-Made use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as backend.
-Used [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make HTTP requests to the backend
-Fetched books from the Bookstore API after the component has rendered
-Refactored  add book and remove book features to persist  changes in the server
- Application behaves the same way after the refactoring

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,40 +1,123 @@
...
+//       ...state,
+//       books: state.books.filter((book) => book.itemID !== action.payload),
+//     }),
+//   },
+// });
+
+// export const { addBook, removeBook } = booksSlice.actions;
+// export default booksSlice.reducer;
+
+/// ////////////////////////////////////////////////////////////////////////////////////////////////
```
Comment (Felix45) [inline src/redux/books/booksSlice.js]: - [ Required ] Please consider removing any commented pieces of code from your project, commented code makes your project look unprofessional and is not considered clean code.

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -1,40 +1,123 @@
...
+      .addCase(removeBook.fulfilled, (state, action) => ({
+        ...state,
+        books: state.books.filter((book) => book.id !== action.payload),
+      }));
   },
 });
 
-export const { addBook, removeBook } = booksSlice.actions;
+// Export Actions and Reducer
+// export const {} = booksSlice.actions;
```
Comment (Felix45) [inline src/redux/books/booksSlice.js]: - [ Required ] Please consider removing any commented pieces of code from your project, commented code makes your project look unprofessional and is not considered clean code.

File: src/redux/books/booksSlice.js [hunk 3]
```diff
@@ -1,40 +1,123 @@
...
+
+    return books;
+  },
+);
+
+export const addBook = createAsyncThunk(
+  'books/addBook',
+  async (book) => {
+    const response = await axios.post('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/d5U1jci9rU4Ou9MD2rVE/books', book);
+    // return response.data;
```
Comment (Felix45) [inline src/redux/books/booksSlice.js]: - [ Required ] Please consider removing any commented pieces of code from your project, commented code makes your project look unprofessional and is not considered clean code.

File: src/components/Booklist/Booklist.js [hunk 4]
```diff
@@ -1,17 +1,23 @@
...
-import { useSelector } from 'react-redux';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import Book from '../Book/Book';
 import Addbook from '../Addbook/Addbook';
+import { fetchBooksList } from '../../redux/books/booksSlice';
 
 export default function Booklist() {
+  const dispatch = useDispatch();
   // const [books] = useState([{ title: 'react', author: 'karan' }]);
```
Comment (Felix45) [inline src/components/Booklist/Booklist.js]: - [ Required ] Please consider removing any commented pieces of code from your project, commented code makes your project look unprofessional and is not considered clean code.

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @karanJ2212 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights :heavy_check_mark: 
- [x] Application works as expected :white_check_mark: 
- [x] All linter checks are passing with green checkmarks :white_check_mark:  
- [x] Repository has a professional readme :white_check_mark:  
- [x] Descriptive pull request title and summary :white_check_mark:  

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand:  @karanJ2212 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights :green_circle: 

- Codebase is clean :+1: 
- Linter checks passed :heavy_check_mark: 

## Optional suggestions

- [ ] It is highly recommended that all links in the README file should be working to create a better user experience. Let's make sure the MIT License link is working. I think the previous reviewer didn't notice this.
- [ ] In the Future Feature section of the README file, you didn't modify the line "Describe 1 - 3 features you will add to the project". This information is from the template hence kindly change it or get rid of the line. Kindly also check similar blockquote in the **Show your support** section

Screenshot from 2023-04-06 07-34-56



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**